diff --git a/node_modules/react-native-contacts/android/src/main/AndroidManifest.xml b/node_modules/react-native-contacts/android/src/main/AndroidManifest.xml
index 9a345fb..009d000 100644
--- a/node_modules/react-native-contacts/android/src/main/AndroidManifest.xml
+++ b/node_modules/react-native-contacts/android/src/main/AndroidManifest.xml
@@ -1,5 +1,4 @@
 <?xml version="1.0" encoding="utf-8"?>
 <manifest
-  xmlns:android="http://schemas.android.com/apk/res/android"
-  package="com.rt2zz.reactnativecontacts" >
+  xmlns:android="http://schemas.android.com/apk/res/android" >
 </manifest>
diff --git a/node_modules/react-native-contacts/ios/RCTContacts/RCTContacts.m b/node_modules/react-native-contacts/ios/RCTContacts/RCTContacts.m
index a303aff..5de088d 100644
--- a/node_modules/react-native-contacts/ios/RCTContacts/RCTContacts.m
+++ b/node_modules/react-native-contacts/ios/RCTContacts/RCTContacts.m
@@ -1,7 +1,7 @@
 #import <AddressBook/AddressBook.h>
 #import <UIKit/UIKit.h>
 #import "RCTContacts.h"
-#import <AssetsLibrary/AssetsLibrary.h>
+#import <Photos/Photos.h>
 #import <React/RCTLog.h>

 @implementation RCTContacts {
@@ -1158,40 +1158,33 @@ + (NSData*) imageData:(NSString*)sourceUri
     }
 }

-enum { WDASSETURL_PENDINGREADS = 1, WDASSETURL_ALLFINISHED = 0};
-
-+ (NSData*) loadImageAsset:(NSURL*)assetURL {
-    //thanks to http://www.codercowboy.com/code-synchronous-alassetlibrary-asset-existence-check/
-
-    __block NSData *data = nil;
-    __block NSConditionLock * albumReadLock = [[NSConditionLock alloc] initWithCondition:WDASSETURL_PENDINGREADS];
-    //this *MUST* execute on a background thread, ALAssetLibrary tries to use the main thread and will hang if you're on the main thread.
-    dispatch_async( dispatch_get_global_queue(DISPATCH_QUEUE_PRIORITY_DEFAULT, 0), ^{
-        ALAssetsLibrary * assetLibrary = [[ALAssetsLibrary alloc] init];
-        [assetLibrary assetForURL:assetURL
-                      resultBlock:^(ALAsset *asset) {
-                          ALAssetRepresentation *rep = [asset defaultRepresentation];
-
-                          Byte *buffer = (Byte*)malloc(rep.size);
-                          NSUInteger buffered = [rep getBytes:buffer fromOffset:0.0 length:rep.size error:nil];
-                          data = [NSData dataWithBytesNoCopy:buffer length:buffered freeWhenDone:YES];
-
-                          [albumReadLock lock];
-                          [albumReadLock unlockWithCondition:WDASSETURL_ALLFINISHED];
-                      } failureBlock:^(NSError *error) {
-                          RCTLog(@"asset error: %@", [error localizedDescription]);
-
-                          [albumReadLock lock];
-                          [albumReadLock unlockWithCondition:WDASSETURL_ALLFINISHED];
-                      }];
++ (NSData *)loadImageAsset:(NSURL *)assetURL {
+    __block NSData *imageData = nil;
+    dispatch_semaphore_t semaphore = dispatch_semaphore_create(0);
+
+    dispatch_async(dispatch_get_global_queue(DISPATCH_QUEUE_PRIORITY_DEFAULT, 0), ^{
+        PHFetchResult *result = [PHAsset fetchAssetsWithALAssetURLs:@[assetURL] options:nil];
+        if (result.count > 0) {
+            PHAsset *asset = result.firstObject;
+            PHImageRequestOptions *options = [[PHImageRequestOptions alloc] init];
+            options.synchronous = YES;
+
+            [[PHImageManager defaultManager] requestImageDataForAsset:asset
+                                                              options:options
+                                                        resultHandler:^(NSData * _Nullable data, NSString * _Nullable dataUTI, UIImageOrientation orientation, NSDictionary * _Nullable info) {
+                imageData = data;
+                dispatch_semaphore_signal(semaphore);
+            }];
+        } else {
+            dispatch_semaphore_signal(semaphore);
+        }
     });

-    [albumReadLock lockWhenCondition:WDASSETURL_ALLFINISHED];
-    [albumReadLock unlock];
+    dispatch_semaphore_wait(semaphore, DISPATCH_TIME_FOREVER);

-    RCTLog(@"asset lookup finished: %@ %@", [assetURL absoluteString], (data ? @"exists" : @"does not exist"));
+    RCTLog(@"asset lookup finished: %@ %@", [assetURL absoluteString], (imageData ? @"exists" : @"does not exist"));

-    return data;
+    return imageData;
 }

 RCT_EXPORT_METHOD(deleteContact:(NSDictionary *)contactData resolver:(RCTPromiseResolveBlock) resolve
